File: python/examples/cp.0.py
#!/usr/bin/python
# Written by andrewt@cse.unsw.edu.au  for COMP2041
# Simple cp implementation using line by line I/O

import sys,os
if len(sys.argv) != 3:
    print >>sys.stderr, "Usage: %s <infile> <outfile>" % sys.argv[0]
    sys.exit(1)
outfile = open(sys.argv[2], 'w')
for line in open(sys.argv[1]):
	outfile.write(line)

File: perl/examples/cp.0.pl
#!/usr/bin/perl -w
# Written by andrewt@cse.unsw.edu.au  for COMP2041
# Simple cp implementation using line by line I/O

die "Usage: $0 <infile> <outfile>\n" if @ARGV != 2;

$infile = shift @ARGV;
$outfile = shift @ARGV;

open(IN, "<$infile") || die("Cannot open $infile: $!");
open(OUT, ">$outfile") || die("Cannot open $outfile: $!");

while ($line = <IN>) {
    print OUT $line;
}

close(IN);
close(OUT);
exit 0;
